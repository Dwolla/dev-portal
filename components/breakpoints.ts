type Breakpoint = "xs" | "sm" | "md" | "lg" | "xl" | "xxl";

// <: extra small devices - portrait phones
const BREAKPOINT_576 = 576;
// > <: small devices - landscape phones
const BREAKPOINT_768 = 768;
// > <: medium devices - tablets
const BREAKPOINT_992 = 992;
// > <: large devices - desktops
const BREAKPOINT_1200 = 1200;
// > <: extra large devices - large desktops, etc.
const BREAKPOINT_1600 = 1600;
// >: extra extra large devices - Big McLargeHuge devices

function breakUpPx(bpt: Breakpoint): number {
  switch (bpt) {
    case "sm":
      return BREAKPOINT_576;
    case "md":
      return BREAKPOINT_768;
    case "lg":
      return BREAKPOINT_992;
    case "xl":
      return BREAKPOINT_1200;
    case "xxl":
      return BREAKPOINT_1600;
    default:
      return 0;
  }
}

function breakDownPx(bpt: Breakpoint): number {
  switch (bpt) {
    case "xs":
      return BREAKPOINT_576;
    case "sm":
      return BREAKPOINT_768;
    case "md":
      return BREAKPOINT_992;
    case "lg":
      return BREAKPOINT_1200;
    case "xl":
      return BREAKPOINT_1600;
    default:
      return 0;
  }
}

export const breakUp = (bpt: Breakpoint) => `min-width: ${breakUpPx(bpt)}px`;
export const breakDown = (bpt: Breakpoint) =>
  `max-width: ${breakDownPx(bpt) - 0.02}px`;
