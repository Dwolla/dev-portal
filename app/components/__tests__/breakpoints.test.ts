import * as breakpoints from "../breakpoints";

test("breakpoints", () => {
  expect(breakpoints).toMatchSnapshot();
});
