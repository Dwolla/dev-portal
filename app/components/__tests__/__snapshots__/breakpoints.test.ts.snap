// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breakpoints 1`] = `
Object {
  "breakDown": [Function],
  "breakUp": [Function],
}
`;
